dss_datamover/tests/test_socket.py

```diff
@@ -33,8 +33,362 @@
 """
 
 import pytest
+import socket
+import os
+import json
+import re
+from socket_communication import ServerSocket, ClientSocket
+from conftest import MockSocket
+import time
 
+def prepare_send_data(msg, header_len, wrong_size=False):
+    if wrong_size:
+        msg_len = (str(len(msg)+3)).zfill(header_len)
+    else:
+        msg_len = (str(len(msg))).zfill(header_len)
+    ret = msg_len + msg
+    return ret
 
-@pytest.mark.usefixtures("get_mock_clientsocket")
+@pytest.mark.usefixtures("get_pytest_configs", "get_config_dict", "get_mock_logger", "get_header_length")
 class TestSocketCommunication():
```

Comment:
Great job on the tests, only recommendation I have is to consolidate them more, meaning that if there are tests that test the positive & negative & corner cases of the same funcitonality, such as socket connect. Then all of those variations should be within that same test case